account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -258,6 +259,105 @@ describe("MA v2 Tests", async () => {
     ).resolves.not.toThrowError();
   });
 
+  it.only("installs native token limit module, then uninstalls", async () => {
+    let provider = (await givenConnectedProvider({ signer })).extend(
+      installValidationActions
+    );
+
+    await setBalance(client, {
+      address: provider.getAddress(),
+      value: parseEther("2"),
+    });
+
+    const spendLimit = parseEther("0.1"); // 0.1 ETH limit
+    const hookInstallData = nativeTokenLimitModule.encodeOnInstallData({
+      entityId: 1,
+      spendLimit,
+    });
+
+    const installResult = await provider.installValidation({
+      validationConfig: {
+        // TODO: Make this a constant "FALLBACK_VALIDATION_CONFIG or something"
+        moduleAddress: zeroAddress,
+        entityId: 0,
+        isGlobal: true,
+        isSignatureValidation: true,
+        isUserOpValidation: true,
+      },
+      selectors: [],
+      installData: "0x",
+      hooks: [
+        {
+          hookConfig: {
+            address: addresses.nativeTokenLimitModule,
+            entityId: 0,
+            hookType: HookType.VALIDATION,
+            hasPreHooks: true,
+            hasPostHooks: false,
+          },
+          initData: hookInstallData,
+        },
+      ],
+    });
+
+    await expect(
+      provider.waitForUserOperationTransaction(installResult)
+    ).resolves.not.toThrowError();
+
+    // Try to send more than the limit - should fail
+    await expect(
+      provider.sendUserOperation({
+        uo: {
+          target: target,
+          value: parseEther("0.2"), // Try to send 0.2 ETH, above the 0.1 limit
+          data: "0x",
+        },
+      })
+    ).rejects.toThrowError();
+
+    // Try to send less than the limit - should succeed
+    const sendResult = await provider.sendUserOperation({
```

Comment:
⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'sendResult' is assigned a value but never used.
<!-- __reviewdog__:ChA0ZjM2OTQ0MmRiNDZkM2JmEgZyZGpzb24= -->